src/firefly/java/edu/caltech/ipac/firefly/server/query/lc/LightCurveProcessor.java

```diff
@@ -47,6 +44,38 @@ public LightCurveProcessor() {
     public DataGroup fetchDataGroup(TableServerRequest request) throws DataAccessException {
 
         PeriodogramAPIRequest req = QueryUtil.assureType(PeriodogramAPIRequest.class, request);
+
+
+        String lightCurveSource = request.getParam(LC_FILE);
+
+        if (isSourceTableFiltered(lightCurveSource)){
```

Comment:
Logic is at the wrong place.  I am not sure if the returned data will be correct.

Reply:
If the table is not filtered, and the table is available to use.  There is no need to save it again.